tests/integration/suite/daprd/conversation/alpha2/grpc/basic.go

```diff
@@ -0,0 +1,248 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	rtv1 "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/suite"
+	"github.com/dapr/kit/ptr"
+)
+
+func init() {
+	suite.Register(new(basic))
+}
+
+type basic struct {
+	daprd *daprd.Daprd
+}
+
+func (b *basic) Setup(t *testing.T) []framework.Option {
+	b.daprd = daprd.New(t, daprd.WithResourceFiles(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: test-alpha2-echo
+spec:
+  type: conversation.echo
+  version: v1
+  metadata:
+  - name: key
+    value: testkey
+`))
+
+	return []framework.Option{
+		framework.WithProcesses(b.daprd),
+	}
+}
+
+func (b *basic) Run(t *testing.T, ctx context.Context) {
+	b.daprd.WaitUntilRunning(t, ctx)
+
+	client := b.daprd.GRPCClient(t, ctx)
+
+	t.Run("all fields", func(t *testing.T) {
+		tool := &rtv1.ConversationTools{
+			ToolTypes: &rtv1.ConversationTools_Function{
+				Function: &rtv1.ConversationToolsFunction{
+					Name:        "test_function",
+					Description: ptr.Of("A test function"),
+					Parameters: map[string]*anypb.Any{
+						"param1": {
+							Value: []byte(`"string"`),
+						},
+					},
+				},
+			},
+		}
+
+		parameters := map[string]*anypb.Any{
+			"max_tokens": {Value: []byte(`100`)},
+			"model":      {Value: []byte(`"test-model"`)},
+		}
+		metadata := map[string]string{
+			"api_key": "test-key",
+			"version": "1.0",
+		}
+
+		contextID := "test-conversation-123"
+		resp, err := client.ConverseAlpha2(ctx, &rtv1.ConversationRequestAlpha2{
+			Name:      "test-alpha2-echo",
+			ContextId: ptr.Of(contextID),
+			// multiple inputs
+			Inputs: []*rtv1.ConversationInputAlpha2{
+				{
+					Messages: []*rtv1.ConversationMessage{
+						{
+							MessageTypes: &rtv1.ConversationMessage_OfUser{
+								OfUser: &rtv1.ConversationMessageOfUser{
+									Name: ptr.Of("test-user"),
+									Content: []*rtv1.ConversationMessageContent{
+										{
+											Text: "well hello there",
+										},
+									},
+								},
+							},
+						},
+					},
+					ScrubPii: ptr.Of(false),
+				},
+				{
+					Messages: []*rtv1.ConversationMessage{
+						{
+							MessageTypes: &rtv1.ConversationMessage_OfSystem{
+								OfSystem: &rtv1.ConversationMessageOfSystem{
+									Name: ptr.Of("test-system"),
+									Content: []*rtv1.ConversationMessageContent{
+										{
+											Text: "You are a helpful assistant",
+										},
+									},
+								},
+							},
+						},
+					},
+					ScrubPii: ptr.Of(true),
+				},
+			},
+			Parameters:  parameters,
+			Metadata:    metadata,
+			ScrubPii:    ptr.Of(true),
```

Comment:
Can we make sure PII data is scrubbed with the Alpha2 method? The Alpha1 tests validate it